lib/worker/entity.js

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+const createEntityFromSubmission = ({ Audits, Entities }, event) => {
+  if (event.details.reviewState === 'approved' && event.details.submissionDefId) {
+    return Entities.processSubmissionDef(event.details.submissionDefId)
+      .then((entity) => {
+        Audits.log({ id: event.actorId }, 'entity.create', { acteeId: entity.aux.dataset.acteeId },
+          { entityUuid: entity.uuid,
+            submissionId: event.details.submissionId,
+            submissionDefId: event.details.submissionDefId });
+      })
+      .catch((problem) => {
```

Comment:
We talked about this more. `Entities.processSubmissionDef()` could fail because of a user error (in which case it'll be an `entityViolation` Problem) or because of a different reason (for example, can't connect to the database). We're thinking:

- If the error is an `entityViolation` Problem, then we log an `entity.create.error` audit log entry.
- If the error isn't an `entityViolation` Problem, then we let the worker do its usual behavior, which is trying to process the event again.
- If the worker somehow fails 5 times, then the user won't ever see an `entity.create.error` audit log entry. It seems like it'd be nice to alert them about this error somehow. But that's something we could do in the future. Matt will file an issue for that.

Reply:
I've filed an issue for the last bullet point at #661.